### PR TITLE
Adjust testers for firebase addition to servlet

### PR DESCRIPTION
**Pull request by dtalton3**
- Change the testers to mock for userID to be sent as headers instead of parameters

### CODE REVIEW

**Comment by kiprot** [comment]
lgtm

**Comment by dtalton3** [comment]
> @googlebot I signed it!

**Comment by dtalton3** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fgoogleinterns%2Fstep147-2020%2Fpull%2F41) for more info**.

**Comment by dtalton3** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fgoogleinterns%2Fstep147-2020%2Fpull%2F41) for more info**.



> @googlebot I signed it!

File: src/main/java/com/google/sps/servlets/MutableHttpServletRequest.java [hunk 1]
```diff
@@ -1,3 +1,5 @@
+package com.google.sps.servlets;
+
```
Comment (octopus1000) [inline src/main/java/com/google/sps/servlets/MutableHttpServletRequest.java:2]: Just curious why do we introduce this?

File: src/main/java/com/google/sps/servlets/ChatroomServlet.java [hunk 2]
```diff
@@ -50,7 +50,7 @@
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-        String userID = request.getParameter("userId");
+        String userID = request.getHeader("userId");
```
Comment (octopus1000) [inline src/main/java/com/google/sps/servlets/ChatroomServlet.java:53]: It this something left out from previous changes? As it's not  a test file.

File: src/main/java/com/google/sps/servlets/MutableHttpServletRequest.java (hunk 1 above)
Comment (dtalton3) [reply src/main/java/com/google/sps/servlets/MutableHttpServletRequest.java:2]: When the auth-token is verified, it gives us a userID. So we need to add that userID to the request as a header
HttpServletRequest doesn't have a addHeader() method so this class creates a HttpServletRequest that we can add the header to. 
https://wilddiary.com/adding-custom-headers-java-httpservletrequest/

File: src/main/java/com/google/sps/servlets/ChatroomServlet.java (hunk 2 above)
Comment (dtalton3) [reply src/main/java/com/google/sps/servlets/ChatroomServlet.java:53]: Yes. It slipped past me

**Pull request merged**